hydra-node/src/Hydra/API/Server.hs

```diff
@@ -117,22 +125,57 @@ runAPIServer ::
   (IsChainState tx) =>
   IP ->
   PortNumber ->
+  Party ->
   Tracer IO APIServerLog ->
   TVar [TimedServerOutput tx] ->
   (ClientInput tx -> IO ()) ->
   TChan (TimedServerOutput tx) ->
   IO ()
-runAPIServer host port tracer history callback responseChannel = do
+runAPIServer host port party tracer history callback responseChannel = do
   traceWith tracer (APIServerStarted port)
   handle onIOException $
     runServer (show host) (fromIntegral port) $ \pending -> do
+      let path = requestPath $ pendingRequest pending
+      queryParams <- uriQuery <$> mkURIBs path
       con <- acceptRequest pending
       chan <- STM.atomically $ dupTChan responseChannel
       traceWith tracer NewAPIConnection
-      forwardHistory con
+
+      -- api client can decide if they want to see the past history of server outputs
+      dontServeHistory <- shouldNotServeHistory queryParams
+      if dontServeHistory
+        then forwardGreetingOnly con
+        else forwardHistory con
+
+      -- api client can decide if they want tx's to be displayed as CBOR instead of plain json
+      txDisplay <- decideOnTxDisplay queryParams
+
       withPingThread con 30 (pure ()) $
-        race_ (receiveInputs con) (sendOutputs chan con)
+        race_ (receiveInputs con) (sendOutputs chan con txDisplay)
  where
+  forwardGreetingOnly con = do
+    time <- getCurrentTime
+    sendTextData con $
+      Aeson.encode
+        TimedServerOutput
+          { time
+          , seq = 0
+          , output = Greetings party :: ServerOutput tx
+          }
+
+  decideOnTxDisplay qp = do
+    k <- mkQueryKey "tx-output"
+    v <- mkQueryValue "cbor"
+    pure $
+      case (QueryParam k v) `elem` qp of
+        True -> TxCBOR
+        False -> TxJSON
+
+  shouldNotServeHistory qp = do
+    k <- mkQueryKey "history"
+    v <- mkQueryValue "0"
+    pure $ (QueryParam k v) `elem` qp
```

Comment:
Could: Not look for a specific value, but look for key `history` and interpret the parameter value where multiple values are negative. (see pascal's comment above)

Reply:
Not using integers anymore